students/YingGuo/session04/Trigrams.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python3
+
+"""Session 04, assignment:Trigrams, student name: Ying Guo"""
+
+words = "I wish I may I wish I might good morning good afternoon hahahahaha lala hahaha".split()
+
+from collections import defaultdict
+import random
+
+def build_trigrams(words):
+
+    trigrams = defaultdict(list)
+
+    for i in range(len(words)-2):
+        pair = tuple(words[i:i+2])
+        follower = words[i+2]
+        trigrams[pair] += [follower]
+    
+    print("This is the Trigrams:")
+    for i in trigrams.keys():
+        print(i, ": ", trigrams[i])
```

Comment:
Minor: Desirable `for` loop iteration for this kind of use case is as follows:
```
for k, v in trigrams.items():
    print(k, ": ", v)
```
At least you don't need to put `.keys()` when you iterate over keys only (so `for i in trigrams:` would have just worked the same). Just FYI, and it'll take some time to absorb all these details...